csrc/tensor_view.cpp

```diff
@@ -1168,15 +1168,39 @@ TensorView* TensorView::cacheFork() {
 TensorView* TensorView::cacheAfter(
     LoadStoreOpType op_type,
     CacheOp cache_op,
-    bool propagate_allocation_domain) {
+    bool propagate_allocation_domain,
+    const std::unordered_set<Expr*>& cached_uses) {
   NVF_ERROR(
       !container()->isA<kir::Kernel>(),
       "Function invalid for kernel container.");
   FusionGuard fg(fusion());
 
+  std::vector<Expr*> target_uses;
+  if (!cached_uses.empty()) {
+    std::unordered_set<Expr*> unique_uses = fusion()->unordered_uses(this);
+    target_uses.reserve(cached_uses.size());
+    for (auto use : cached_uses) {
+      NVF_ERROR(
+          unique_uses.count(use),
+          "cached_uses is not among the use of the TensorView");
+      target_uses.push_back(use);
```

Comment:
Isn't this ordering still non deterministic? I think the parameter itself needs to be deterministically ordered.

Reply:
Good catch. Sorry I missed the `cached_uses`. Let me give it another try.